### PR TITLE
Tags added with views and moved articles#index list to a partial to reuse for tags list

### PR DESCRIPTION
**Pull request by coolRaider**
**What this PR does:**
- integrated acts_as_taggable_on
- added controller and views for the tags with appropriate routes
- moved articles#index list to a partial to reuse for showing articles with a particular tag when clicked on from a tag link in article#show path.

### CODE REVIEW

File: app/views/acts_as_taggable_on/tags/_tag.html.haml [hunk 1]
```diff
@@ -0,0 +1 @@
+= link_to tag.name, tag_path(tag)
```
Comment (joney000) [inline app/views/acts_as_taggable_on/tags/_tag.html.haml:1]: a new line at end.

File: app/views/tags/index.html.haml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+%h1 Tags
+= render @tags
```
Comment (joney000) [inline app/views/tags/index.html.haml:2]: a new line at EOF.

File: .rake_tasks~ [hunk 3]
```diff
@@ -0,0 +1,43 @@
+about
```
Comment (tarunjadhwani) [inline .rake_tasks~:1]: check if this can be added to gitignore

File: config/routes.rb [hunk 4]
```diff
@@ -34,17 +34,22 @@
 
 Rails.application.routes.draw do
 
+  get 'tags/index'
```
Comment (tarunjadhwani) [inline config/routes.rb]: not needed as these would be generated with resources.

**Comment by tarunjadhwani** [comment]
@coolRaider resolve signin signup bug

**Comment by coolRaider** [comment]
sign in, sign up was not a bug, the environment variables weren't set on the machine for username and password, put them in the .bashrc file, so now they are working.

File: config/routes.rb (hunk 4 above)
Comment (coolRaider) [inline config/routes.rb]: removed on merge with search branch

File: .rake_tasks~ (hunk 3 above)
Comment (coolRaider) [inline .rake_tasks~:1]: removed from repo and put in .ignore file.

File: app/views/tags/index.html.haml (hunk 2 above)
Comment (coolRaider) [inline app/views/tags/index.html.haml:2]: done.

File: app/views/acts_as_taggable_on/tags/_tag.html.haml (hunk 1 above)
Comment (coolRaider) [inline app/views/acts_as_taggable_on/tags/_tag.html.haml:1]: done.

**Pull request merged**